expression/date_arith.go

```diff
@@ -137,3 +164,19 @@ func (da *DateArith) evalArgs(ctx context.Context, args map[interface{}]interfac
 
 	return t, years, months, days, durations, nil
 }
+
+func (da *DateArith) evalDaysForm(val interface{}) (interface{}, error) {
+	switch val.(type) {
```

Comment:
switch v := val.(type)

so you can use v directly later, no need to write val.(string).